.github/workflows/build-docker.yaml

```diff
@@ -0,0 +1,62 @@
+# This GitHub Actions workflow builds a Docker image for the
+# cfa-epinow2-pipeline-docker project. It consists of two jobs: build_image
+# and run_tests:
+#
+# - The build_image job checks out the code and outputs the IMAGE_NAME
+#   environment variable.
+#
+# - The run_tests job runs on the ubuntu-latest runner and uses the Docker image
+#   built in the build_image job.
+#
+#   It checks out the code, builds the package using R CMD build, and checks the
+#   package using R CMD check.
+
+name: Builds Docker image
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+env:
+  IMAGE_NAME: gvegayon/cfa-epinow2-pipeline
+  REGISTRY: docker.io
+
+jobs:
+  build_image_dependencies:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Check cache
+        uses: actions/cache@v3
+        id: cache
+        with:
+          key: docker-dependencies-${{ runner.os }}-${{ hashFiles('./DESCRIPTION', './Dockerfile-dependencies') }}
+          path:
+            ./DESCRIPTION
+
+      - name: Build and push
+        uses: docker/build-push-action@v6
+        if: steps.cache.outputs.cache-hit != 'true'
+        with:
+          push: false
+          tags: ${{ env.REGISTRY}}/${{ env.IMAGE_NAME }}:latest
+          file: ./Dockerfile-dependencies
+
+    outputs:
+      IMAGE_NAME: ${{ env.REGISTRY}}/${{ env.IMAGE_NAME }}:latest
+
+  build_image:
+    needs: build_image_dependencies
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Build and push
+        uses: docker/build-push-action@v6
+        with:
+          push: false
+          tags: ${{ needs.build_image_dependencies.outputs.IMAGE_NAME }}
+          file: ./Dockerfile
```

Comment:
Does this also need a cache step? What will happen in PRs that are e.g. just a readme update with no impact on the R package?

I'm open to the idea that it's simpler/better to always refresh the image on merge to main and not try to be clever about detecting changes that are/aren't to the package. But it would be good to document either way what's happening here because I'm having some trouble figuring it out!

Reply:
We could cache other bits, but right now, I think it would be overkill. I would suggest making that an issue when the build takes longer.